hydrator-common/src/main/java/co/cask/hydrator/common/ReferenceBatchSink.java

```diff
@@ -38,8 +38,10 @@ public ReferenceBatchSink(ReferencePluginConfig config) {
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     super.configurePipeline(pipelineConfigurer);
-    // Verify that reference name meets dataset id constraints
-    @SuppressWarnings("unused")
-    DatasetId datasetId = new DatasetId(NamespaceId.DEFAULT.getNamespace(), config.referenceName);
+    if (!config.containsMacro(Constants.Reference.REFERENCE_NAME)) {
+      // Verify that reference name meets dataset id constraints
+      @SuppressWarnings("unused")
+      DatasetId datasetId = new DatasetId(NamespaceId.DEFAULT.getNamespace(), config.referenceName);
```

Comment:
If `referenceName` contains a macro, does this need to be (conditionally) done at runtime?